ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/CollectionSelection.tsx

```diff
@@ -86,25 +103,18 @@ function CollectionSelection({
     const sortedCollections = useMemo(() => {
         const availableScopes: CollectionSlim[] = [...data.flat(), ...createdCollections];
 
+        // Add the individual pre-selected collection to the list of available scopes. If it already
+        // exists in the list, it will be uniq'd out.
+        if (selectedScopeDetails) {
+            availableScopes.push(selectedScopeDetails.collection);
+        }
+
         // This is inefficient due to the multiple loops and the fact that we are already tracking
         // uniqueness for the _server side_ values, but need to do it twice to handle possible client
         // side values. However, 'N' should be small here and we are memoizing the result.
         const sorted = sortBy(availableScopes, ({ name }) => name.toLowerCase());
         return uniqBy(sorted, 'id');
-    }, [data, createdCollections]);
-
-    // This makes sure that if a collection was deleted then we clear the scopeId
-    useEffect(() => {
-        if (!isFetchingNextPage) {
-            const selectedCollection = sortedCollections.find(
-                (collection) => collection.id === selectedScope?.id
-            );
-            if (!selectedCollection) {
-                onChange(null);
-            }
-        }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [data, isFetchingNextPage]);
```

Comment:
This effect previously reset the "scope" portion of the field when it detected that the current selected scope was not part of the client side collection list for the dropdown. @sachaudh correct me if I am wrong, but I believe this is a safety check from when the backend did not restrict collection deletion.

As it functions today these are the following scenarios:
- The selected collection scope for the report is deleted after the report is saved (**not possible due to BE check**)
- The selected collection is changed via the form, and deleted from the collection page before the report is saved. In this case, the form field will not be reset because the collection data has already been loaded client side on the report page and it does not know of the deletion. Upon trying to save the report, an error will be shown at the top of the UI.
- The selected collection is not present in the initial page of data, so is cleared from the form. (**current breaking behavior**)

Reply:
True, we can't delete a collection that is currently being used. I was a little confused by the second point though. If you haven't saved the report yet, it'll still prevent you from deleting the collection.

<img width="572" alt="Screenshot 2023-11-20 at 2 29 17 PM" src="https://github.com/stackrox/stackrox/assets/4805485/1f7a862c-21b9-4b7c-8dc6-77d6b520b710">

Reply:
But overall, this works 👍🏼